crates/oxc_linter/src/rules/jest/padding_around_test_blocks.rs

```diff
@@ -0,0 +1,246 @@
+    ///
+    /// it('bar', () => {});
+    /// ```
+    PaddingAroundTestBlocks,
+    jest,
+    style,
+    fix
+);
+
+enum NodeType {
+    Test,
+    Statement,
+}
+
+impl Rule for PaddingAroundTestBlocks {
+    fn run_once(&self, ctx: &LintContext<'_>) {
+        let mut nodes = collect_possible_jest_call_node(ctx)
+            .iter()
+            .filter_map(|possible_jest_node| {
+                let node = possible_jest_node.node;
+                let AstKind::CallExpression(call_expr) = node.kind() else {
+                    return None;
+                };
+                let jest_fn_call = parse_general_jest_fn_call(call_expr, possible_jest_node, ctx)?;
+                let ParsedGeneralJestFnCall { kind, .. } = &jest_fn_call;
+                let kind = kind.to_general()?;
+                if kind != JestGeneralFnKind::Test {
+                    return None;
+                }
+                Some((NodeType::Test, node))
+            })
+            .collect_vec();
+        for node in ctx.nodes() {
+            if node.kind().is_statement() {
+                nodes.push((NodeType::Statement, node));
+            }
+        }
+        nodes.sort_by_key(|(_, node)| node.span().end);
+        let mut prev_node: Option<&AstNode<'_>> = None;
+        for (node_type, node) in nodes {
+            match node_type {
+                NodeType::Test => {
+                    if let Some(prev_node) = prev_node {
+                        if prev_node.span().end > node.span().start {
+                            continue;
+                        }
+                        let mut comments_range =
+                            ctx.comments_range(prev_node.span().end..node.span().start);
+                        let mut span_between_start = prev_node.span().end;
+                        let mut span_between_end = node.span().start;
+                        if let Some(last_comment_span) =
+                            comments_range.next_back().map(|comment| comment.span)
+                        {
+                            let space_after_last_comment = ctx
+                                .source_range(Span::new(last_comment_span.end, node.span().start));
+                            let space_before_last_comment = ctx.source_range(Span::new(
+                                prev_node.span().end,
+                                last_comment_span.start,
+                            ));
+                            if space_after_last_comment.matches('\n').count() > 1
+                                || space_before_last_comment.matches('\n').count() == 0
+                            {
+                                span_between_start = last_comment_span.end;
+                            } else {
+                                span_between_end = last_comment_span.start;
+                            }
+                        }
+                        let span_between = Span::new(span_between_start, span_between_end);
```

Comment:
`span_between` is only needed when the user is using the `fixer`.
Please move all logic inside the `|fixer| {}` function. So we do not execute it, when it is not needed.